definitions/features/hammer.rb

```diff
@@ -199,7 +199,11 @@ def load_from_file(file_path)
 
   def password_from_answers
     return nil unless feature(:installer)
-    feature(:installer).answers['foreman']['admin_password']
+    if check_min_version('foreman', '1.22')
+      feature(:installer).answers['foreman']['initial_admin_password']
```

Comment:
In latest version, installer uses `initial_admin_username and initial_admin_password`.
Foreman Maintain loads hammer configurations first. If those are invalid then we have [hardcoded](https://github.com/theforeman/foreman_maintain/blob/6b342a5d10186a4e820f95f31899d3a789ba2a57/definitions/features/hammer.rb#L31) username to `admin`. 
I think it would be good if it considers `initial_admin_password` only when initial_admin_username matches with username from foreman_maintain's config.

@iNecas, @mbacovsky, any thoughts?

Reply:
Good call

Reply:
@kgaikwad updated PR according to your suggestions.

Reply:
@kgaikwad had good point. I'd just suggest not to use username form the config for comparison (currently it will be always 'admin') but instead use new `feature(:foreman_server).admin_username`.